---
title: "Upozornění kompilátoru (úroveň 1) CS1058"
ms.date: 07/20/2015
ms.prod: .net
ms.technology: devlang-csharp
ms.topic: error-reference
f1_keywords: CS1058
helpviewer_keywords: CS1058
ms.assetid: ed50590c-f130-47c3-976d-322a6c8f996d
caps.latest.revision: "11"
author: BillWagner
ms.author: wiwagn
ms.openlocfilehash: 3d5d32b17c9803c4df82ff4e8cc3b0672a386b78
ms.sourcegitcommit: bd1ef61f4bb794b25383d3d72e71041a5ced172e
ms.translationtype: MT
ms.contentlocale: cs-CZ
ms.lasthandoff: 10/18/2017
---
# <a name="compiler-warning-level-1-cs1058"></a><span data-ttu-id="e22d2-102">Upozornění kompilátoru (úroveň 1) CS1058</span><span class="sxs-lookup"><span data-stu-id="e22d2-102">Compiler Warning (level 1) CS1058</span></span>
<span data-ttu-id="e22d2-103">Předchozí klauzule catch již zachytí všechny výjimky.</span><span class="sxs-lookup"><span data-stu-id="e22d2-103">A previous catch clause already catches all exceptions.</span></span> <span data-ttu-id="e22d2-104">Všechny výjimky vydané bude zabaleny v System.Runtime.CompilerServices.RuntimeWrappedException</span><span class="sxs-lookup"><span data-stu-id="e22d2-104">All exceptions thrown will be wrapped in a System.Runtime.CompilerServices.RuntimeWrappedException</span></span>  
  
 <span data-ttu-id="e22d2-105">Tento atribut zapříčiní CS1058 pokud `catch()` bloku nemá žádný typ po zadané výjimky `catch (System.Exception e)` bloku.</span><span class="sxs-lookup"><span data-stu-id="e22d2-105">This attribute causes CS1058 if a `catch()` block has no specified exception type after a `catch (System.Exception e)` block.</span></span> <span data-ttu-id="e22d2-106">Upozornění informuje o tom, který `catch()` bloku nezachytí žádné výjimky.</span><span class="sxs-lookup"><span data-stu-id="e22d2-106">The warning advises that the `catch()` block will not catch any exceptions.</span></span>  
  
 <span data-ttu-id="e22d2-107">A `catch()` blokovat po `catch (System.Exception e)` bloku může zachytit výjimky specifikací CLS, pokud `RuntimeCompatibilityAttribute` je nastaven na hodnotu false v souboru AssemblyInfo.cs: `[assembly: RuntimeCompatibilityAttribute(WrapNonExceptionThrows = false)]`.</span><span class="sxs-lookup"><span data-stu-id="e22d2-107">A `catch()` block after a `catch (System.Exception e)` block can catch non-CLS exceptions if the `RuntimeCompatibilityAttribute` is set to false in the AssemblyInfo.cs file: `[assembly: RuntimeCompatibilityAttribute(WrapNonExceptionThrows = false)]`.</span></span> <span data-ttu-id="e22d2-108">Pokud tento atribut není explicitně nastavena na hodnotu false, jsou všechny výjimky vyvolané bez podpory CLS zabaleny jako výjimky a `catch (System.Exception e)` je zachytí bloku.</span><span class="sxs-lookup"><span data-stu-id="e22d2-108">If this attribute is not set explicitly to false, all thrown non-CLS exceptions are wrapped as Exceptions and the `catch (System.Exception e)` block catches them.</span></span> <span data-ttu-id="e22d2-109">Další informace najdete v tématu [postupy: zachycení bez výjimek specifikací CLS -](../../../csharp/programming-guide/exceptions/how-to-catch-a-non-cls-exception.md).</span><span class="sxs-lookup"><span data-stu-id="e22d2-109">For more information, see [How to: Catch a non-CLS Exception](../../../csharp/programming-guide/exceptions/how-to-catch-a-non-cls-exception.md).</span></span>  
  
## <a name="example"></a><span data-ttu-id="e22d2-110">Příklad</span><span class="sxs-lookup"><span data-stu-id="e22d2-110">Example</span></span>  
 <span data-ttu-id="e22d2-111">Následující příklad generuje CS1058.</span><span class="sxs-lookup"><span data-stu-id="e22d2-111">The following example generates CS1058.</span></span>  
  
```csharp  
// CS1058.cs  
// CS1058 expected  
using System.Runtime.CompilerServices;  
  
// the following attribute is set to true by default in the C# compiler  
// set to false in your source code to resolve CS1058  
[assembly: RuntimeCompatibilityAttribute(WrapNonExceptionThrows = true)]  
  
class TestClass   
{  
   static void Main()   
   {  
      try {}  
  
      catch (System.Exception e) {   
         System. Console.WriteLine("Caught exception {0}", e);  
      }  
  
      catch {}   // CS1058. This line will never be reached.  
   }  
}  
```
